### DISCUSSION TITLE
uploading using esptool

### DISCUSSION THREAD

**Discussion by netzbasteln** [ohAnd/dtuGateway#46] [Show and tell]
I was able to upload the binary using esptool.py instead of the windows-only app by espressif.

`esptool.py --port /dev/cu.wchusbserial210 --baud 921600 write_flash --flash_size 4MB --flash_mode dio 0 dtuGateway_release_1.6.0009.bin`

**Comment by skyynet** [comment]
I used the esptool to flash the ESP32 on my mac. Seemed to work.

```
esptool.py --port /dev/cu.SLAB_USBtoUART --baud 921600 write_flash --flash_size 4MB --flash_mode dio 0 dtu.bin
esptool.py v4.8.1
Serial port /dev/cu.SLAB_USBtoUART
Connecting....
Detecting chip type... Unsupported detection protocol, switching and trying again...
Connecting....
Detecting chip type... ESP32
Chip is ESP32-D0WD-V3 (revision v3.1)
Features: WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Crystal is 40MHz
MAC: 3c:8a:1f:77:65:c4
Uploading stub...
Running stub...
Stub running...
Changing baud rate to 921600
Changed.
Configuring flash size...
Flash will be erased from 0x00000000 to 0x00162fff...
Compressed 1452752 bytes to 866396...
Wrote 1452752 bytes (866396 compressed) at 0x00000000 in 15.9 seconds (effective 728.9 kbit/s)...
Hash of data verified.

Leaving...
Hard resetting via RTS pin...
```

I assumed, I'd find a new Wi-Fi network from the ESP's access point afterwards, but nothing.
Any idea?

**Reply by ohAnd** [reply]
can you show your serial log after rebooting ... there should be shown that the AP was started or if there are any general errors.

Usually your assumption is right... after first flashing dtuGateway is starting in "factory mode" and starting an own AP for first setup.
